src/Components/Components/test/Auth/AuthorizeRouteViewTest.cs

```diff
@@ -0,0 +1,131 @@
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Components.RenderTree;
+using Microsoft.AspNetCore.Components.Test.Helpers;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Components
+{
+    public class AuthorizeRouteViewTest
+    {
+        private readonly static IReadOnlyDictionary<string, object> EmptyParametersDictionary = new Dictionary<string, object>();
+        private readonly TestAuthenticationStateProvider _authenticationStateProvider;
+        private readonly TestRenderer _renderer;
+        private readonly RouteView _authorizeRouteViewComponent;
+        private readonly int _authorizeRouteViewComponentId;
+
+        public AuthorizeRouteViewTest()
+        {
+            _authenticationStateProvider = new TestAuthenticationStateProvider();
+            _authenticationStateProvider.CurrentAuthStateTask = Task.FromResult(
+                new AuthenticationState(new ClaimsPrincipal(new ClaimsIdentity())));
+
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddSingleton<AuthenticationStateProvider>(_authenticationStateProvider);
+            serviceCollection.AddSingleton<IAuthorizationPolicyProvider, TestAuthorizationPolicyProvider>();
+            serviceCollection.AddSingleton<IAuthorizationService, TestAuthorizationService>();
+
+            _renderer = new TestRenderer(serviceCollection.BuildServiceProvider());
+            _authorizeRouteViewComponent = new AuthorizeRouteView();
+            _authorizeRouteViewComponentId = _renderer.AssignRootComponentId(_authorizeRouteViewComponent);
+        }
+
+        [Fact]
+        public void WithoutCascadedAuthenticationState_WrapsOutputInCascadingAuthenticationState()
+        {
+            // Arrange/Act
+            var routeData = new ComponentRouteData(typeof(TestPageComponent), EmptyParametersDictionary);
+            _renderer.RenderRootComponent(_authorizeRouteViewComponentId, ParameterView.FromDictionary(new Dictionary<string, object>
+            {
+                { nameof(AuthorizeRouteView.RouteData), routeData }
+            }));
+
+            // Assert
+            var batch = _renderer.Batches.Single();
+            var componentInstances = batch.ReferenceFrames
+                .Where(f => f.FrameType == RenderTreeFrameType.Component)
+                .Select(f => f.Component);
+
+            Assert.Collection(componentInstances,
+                // This is the hierarchy inside the AuthorizeRouteView, which contains its
+                // own CascadingAuthenticationState
+                component => Assert.IsType<CascadingAuthenticationState>(component),
+                component => Assert.IsType<CascadingValue<Task<AuthenticationState>>>(component),
+                component => Assert.True(component is AuthorizeViewCore),
```

Comment:
Assert.IsAssignableTo (or similar name) does this with a better error message.

Reply:
I actually looked for such an assertion when first writing this, but there isn't one. I saw there is an `IsAssignableFrom`, which sounded like the opposite of what I wanted, so didn't use it.

Turns out that `IsAssignableFrom` actually means what you and I both think should be called `IsAssignableTo`, so have switched to use that.